Summary

Wikitelmex opened the pull request:
- follow the layout of the wireframes provided. personalized the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
    -  built with Devise.
- Food list:
    - display a list of food as in the wireframe.
    - lead to the form that allows users to add new food.
- Recipies list:
    - display a list of recipes created by the logged-in user as in the wireframe.
    - lead to recipe details.
    - If the user is the owner of the recipe, should allow user to delete it.
- Public recipe list:
    - display a list of all public recipes ordered by newest as in the wireframe.
    - lead to recipe details.
- Recipe details:
    - If the recipe is public or the user is the owner of the recipe, display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, lead to the form that allows user to add new food.

wikitelmex1 committed:
initial commit

wikitelmex1 committed:
configuration devise

wikitelmex1 committed:
Users model
devise views
devise controllers
login view
singup view

peterdgreat committed:
Merge pull request #30 from Wikitelmex/feature/authentication

Feature/authentication

wikitelmex1 committed:
# Recipe model and Migration
- create recipe model with validators and relationships
- add user model relationship to recipes
- create migration
- implement migration to database and schema

wikitelmex1 committed:
# navbar
- add some required styles to navbar
- add JS bundles required for dropdown menus
- create navbar partial to be redered when it's necessary

wikitelmex1 committed:
# public recipes
Note: taked to test bootstrap features
- add public_recipes controller and index view
- add public_recipes route

wikitelmex1 committed:
fix linters

wikitelmex1 committed:
- add working link to home
- add working link to public recipes

peterdgreat committed:
Food Model Migration and Controller Implemented

peterdgreat committed:
Fixed Linters

wikitelmex1 committed:
Merge pull request #31 from Wikitelmex/feature/recipe

Feature/recipe

peterdgreat committed:
Merge branch 'develop' of github.com:Wikitelmex/recipe-project into feature/food-model-migration-views

peterdgreat committed:
Stled Foods, Updated price type, implemented delete method

wikitelmex1 committed:
Merge pull request #32 from Wikitelmex/feature/food-model-migration-views

Food Model Migration and Controller Implemented

wikitelmex1 committed:
# ALERT
direct changes pushed to develop

- recipefood model
- recipefood validations
- recipefood relationships
- recipefood migrations

wikitelmex1 committed:
minor improvements
-show user name in navbar
- fix price in food view to accept decimal values

peterdgreat committed:
Recipe Food View and Controller Added

wikitelmex1 committed:
# summary
- fix html format for  food
- modify migrations to have cooking and preparation times as integer for minutes
- add recipes controler index, show, destroy, new, create
- add recipes views index, show, destroy, new, create

peterdgreat committed:
Merge pull request #33 from Wikitelmex/feature/RecipeFood-Controller-View

Recipe Food View and Controller Added

wikitelmex1 committed:
- add n+1 solution to recipe, recipe_food, food
- add ingredients controller to support Peter issues with electricity, must be recipe_food controller
- add ingredients views
- add routes for ingredients

wikitelmex1 committed:
# summary
- add edit ingredients features
- include jquery library
- add sorting to table in recipe#show
- add edit routes

wikitelmex1 committed:
- shopping list feature

wikitelmex1 committed:
- public recipes improvements
- logout feature
- add info to home page
- restriction to no owners in public recipes

wikitelmex1 committed:
- fix linters

wikitelmex1 committed:
- fix add recipe issue

wikitelmex1 committed:
- add order by in public recipes to fix requirements

wikitelmex1 committed:
fix linters

wikitelmex1 committed:
Merge pull request #34 from Wikitelmex/feature/recipecv

Feature/recipecv

wikitelmex1 committed:
## Summary
- add cancancan gem
- authenticate and authorize foods controller
- authenticate and authorize recipes controller
- deauthenticate and deauthorize home controller
- deauthenticate and deauthorize public_recipes controller
- fix food new action, before: user = first, after: user = curren_user
- fix recipes controller,, before: local object, after: recipe_params.permit
- add cancancan ability with authorization
- fix minimal style issues
- add column to user, column: role
- add/execute migration

wikitelmex1 committed:
- fix linters

wikitelmex1 committed:
improve readme

wikitelmex1 committed:
Merge pull request #35 from Wikitelmex/feature/authorization

Feature/authorization

t-yanick commented on app/controllers/users/registrations_controller.rb:
```diff
@@ -0,0 +1,79 @@
+class Users::RegistrationsController < Devise::RegistrationsController
+  # before_action :configure_sign_up_params, only: [:create]
+  # before_action :configure_account_update_params, only: [:update]
+
+  # GET /resource/sign_up
+  # def new
+  #   super
+  # end
```
- [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

t-yanick commented on app/controllers/users/registrations_controller.rb:
```diff
@@ -0,0 +1,79 @@
+  # The path used after sign up for inactive accounts.
+  # def after_inactive_sign_up_path_for(resource)
+  #   super(resource)
+  # end
```
- [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

Wikitelmex replied:
done

t-yanick commented on app/controllers/users/sessions_controller.rb:
```diff
@@ -0,0 +1,37 @@
+class Users::SessionsController < Devise::SessionsController
+  # before_action :configure_sign_in_params, only: [:create]
+
+  # GET /resource/sign_in
+  # def new
+  #   super
+  # end
+
+  # POST /resource/sign_in
```
- [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

Wikitelmex replied:
done

t-yanick commented on app/controllers/users/sessions_controller.rb:
```diff
@@ -0,0 +1,37 @@
+  end
+
+  # protected
+
+  # If you have extra params to permit, append them to the sanitizer.
+  # def configure_sign_in_params
+  #   devise_parameter_sanitizer.permit(:sign_in, keys: [:attribute])
+  # end
```
- [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

t-yanick commented on app/models/user.rb:
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
- [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

t-yanick commented on db/migrate/20220207193241_devise_create_users.rb:
```diff
@@ -0,0 +1,44 @@
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
```
- [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

t-yanick commented on spec/helpers/home_helper_spec.rb:
```diff
@@ -0,0 +1,15 @@
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
```
- [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

t-yanick commented on spec/helpers/ingredients_helper_spec.rb:
```diff
@@ -0,0 +1,15 @@
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
```
- [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

t-yanick commented on spec/helpers/public_recipes_helper_spec.rb:
```diff
@@ -0,0 +1,15 @@
+# end
+RSpec.describe PublicRecipesHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
- [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

t-yanick commented on spec/helpers/recipes_helper_spec.rb:
```diff
@@ -0,0 +1,15 @@
+# end
+RSpec.describe RecipesHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
- [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

t-yanick reviewed (changes requested):
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

wikitelmex1 committed:
# changes required 1
- remove comments from registrations
- remove comments from sessions
- remove comments from user
- remove comments from migration user
- remove comments from spec helpers

wikitelmex1 committed:
# changes required 1 (proactive taks)
- delete comments in foods_controller
- delete comments in  ingredients_controller
- delete comments in recipe_foods_controller
- delete comments in ability model
- delete comments in config routes
- delete comments in devise_create_users migration

t-yanick reviewed (approved):
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
